src/megaapi_impl.cpp

```diff
@@ -13266,27 +13296,46 @@ void MegaApiImpl::syncupdate_local_lockretry(bool waiting)
 }
 #endif
 
-void MegaApiImpl::backupput_result(const Error&, handle backupId)
+void MegaApiImpl::backupput_result(const Error& e, handle backupId)
 {
     mHeartBeatMonitor->digestPutResult(backupId);
+    if (requestMap.find(client->restag) == requestMap.end()) return;
+    MegaRequestPrivate* request = requestMap.at(client->restag);
+    if (!request || (request->getType() != MegaRequest::TYPE_BACKUP_PUT)) return;
+
+    fireOnRequestFinish(request, make_unique<MegaErrorPrivate>(e));
 }
 
-void MegaApiImpl::backupupdate_result(const Error&, handle /*backupId*/)
+void MegaApiImpl::backupupdate_result(const Error& e, handle /*backupId*/)
 {
+    if (requestMap.find(client->restag) == requestMap.end()) return;
+    MegaRequestPrivate* request = requestMap.at(client->restag);
+    if (!request || (request->getType() != MegaRequest::TYPE_BACKUP_PUT)) return;
 
+    fireOnRequestFinish(request, make_unique<MegaErrorPrivate>(e));
 }
 
-void MegaApiImpl::backupputheartbeat_result(const Error&)
+void MegaApiImpl::backupputheartbeat_result(const Error& e)
 {
+    if (requestMap.find(client->restag) == requestMap.end()) return;
+    MegaRequestPrivate* request = requestMap.at(client->restag);
+    if (!request || (request->getType() != MegaRequest::TYPE_BACKUP_PUT_HEART_BEAT)) return;
 
+    fireOnRequestFinish(request, make_unique<MegaErrorPrivate>(e));
 }
 
 void MegaApiImpl::backupremove_result(const Error& e, handle backupId)
 {
     if (e || backupId != UNDEF)
     {
         LOG_err << "backupremove_result failed: " << MegaError::getErrorString(e);
+        return;
```

Comment:
Request have to end (fireOnRequestFinish) even if they fail

Reply:
removed the return statement